### PR TITLE
Readds  	Bats, Loot spawners, and removing a depricated flag from an object 	

### PR DESCRIPTION
**Pull request by TDcoolguy300**
Readds missing crap

### CODE REVIEW

**Comment by JohnGinnane** [comment]
If we're using `STOPSPRESSUREDMAGE` instead of `BLOCKHAIR` then you'll need to update a few of our files to make sure they're using the new one:

https://github.com/HippieStationCode/HippieStation13/search?p=2&q=BLOCKHAIR&utf8=%E2%9C%93

Some might just be our code, so not present in /tg/

**Comment by TDcoolguy300** [comment]
@JohnGinnane STOPSPRESSUREDMAGE is a flag that does just that, it stops pressure damage. It appears to exist as proper in the rebase. Remember, the HIDEHAIR flag in the flag_inv variable handles hiding the hair.

File: code/game/objects/items/weapons/baseball.dm [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+	w_class = 1.0
+	force = 0
+	throwforce = 5
+	throw_range = 7
+
+/obj/item/weapon/baseballbat/suicide_act(mob/user)
+	user.visible_message("<span class='suicide'>[user] smashes the baseball bat into \his head! It looks like \he's trying to commit suicide..</span>")
+	return (BRUTELOSS)
+
+/mob/living/carbon/human/hitby(atom/movable/AM, skipcatch, hitpush = 1, blocked = 0, zone)
```
Comment (JohnGinnane) [inline code/game/objects/items/weapons/baseball.dm:70]: Shouldn't this be in human defences? It also seems snowflakey to just handle baseballs

File: code/game/objects/items/weapons/baseball.dm [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+	var/armor = getarmor(check_zone(zone), "melee")
+	if(armor >= 100) return
+	if(zone == "head" && I.throwforce >= 6) //This is kind of a terrible way to check if the baseball was batted but whatever
+		if(stat == CONSCIOUS && prob(50)) //decent chance to make up for the already-RNG zone picking
+			if(armor < 40) //It only KO's you if you don't have head armor
+				visible_message("<span class='danger'>[src] has been knocked unconscious!</span>", \
+								"<span class='userdanger'>[src] has been knocked unconscious!</span>")
+				apply_effect(6, PARALYZE, armor) //Since it's ranged, we don't want to make KO too OP
+				if(prob(50))
+					ticker.mode.remove_revolutionary(mind)
```
Comment (JohnGinnane) [inline code/game/objects/items/weapons/baseball.dm:86]: Wow... Just wow, I didn't know this was possible

File: code/game/objects/items/weapons/baseball.dm [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+/mob/living/carbon/human/hitby(atom/movable/AM, skipcatch, hitpush = 1, blocked = 0, zone)
+	..()
+	var/obj/item/baseball/I = AM
+	if(!istype(I))
+		return
+	if(zone == "")
+		zone = ran_zone("chest", 65)
+	var/armor = getarmor(check_zone(zone), "melee")
+	if(armor >= 100) return
+	if(zone == "head" && I.throwforce >= 6) //This is kind of a terrible way to check if the baseball was batted but whatever
```
Comment (JohnGinnane) [inline code/game/objects/items/weapons/baseball.dm:79]: 'I' could be any object really, I don't think just baseballs

File: code/modules/clothing/head/misc.dm [hunk 4]
```diff
@@ -169,7 +169,7 @@
 	desc = "A helmet made out of chitinous alien hide."
 	flags_inv = HIDEMASK|HIDEEARS|HIDEEYES|HIDEFACE|HIDEHAIR|HIDEFACIALHAIR
 	alternate_screams = list('sound/voice/hiss1.ogg','sound/voice/hiss2.ogg','sound/voice/hiss3.ogg','sound/voice/hiss4.ogg','sound/voice/hiss5.ogg','sound/voice/hiss6.ogg')
-	flags = BLOCKHAIR | STOPSPRESSUREDMAGE
+	flags = STOPSPRESSUREDMAGE
```
Comment (JohnGinnane) [inline code/modules/clothing/head/misc.dm:172]: Shouldn't HIDEHAIR be here as well?

File: code/game/objects/items/weapons/baseball.dm (hunk 1 above)
Comment (JohnGinnane) [reply code/game/objects/items/weapons/baseball.dm:70]: Okay I did some reading on ..() and this is normal

File: code/game/objects/items/weapons/baseball.dm (hunk 3 above)
Comment (JohnGinnane) [reply code/game/objects/items/weapons/baseball.dm:79]: I think ignore my comment here as this is the proc for handling what happens when baseball hits something

**Comment by JohnGinnane** [comment]
Made those comments a few days ago, didn't realise there was this new review system in place

File: code/modules/clothing/head/misc.dm (hunk 4 above)
Comment (TDcoolguy300) [reply code/modules/clothing/head/misc.dm:172]: Look at bloody FLAGS_INV, it's handled there

**Review by JohnGinnane — APPROVED** [review]
Looks okay

**Pull request merged**